cJSON.c

```diff
@@ -1756,7 +1756,7 @@ CJSON_PUBLIC(int) cJSON_GetArraySize(const cJSON *array)
     return (int)size;
 }
 
-static cJSON* get_array_item(const cJSON *array, size_t index)
+static cJSON* get_array_item(const cJSON *array, size_t _index)
```

Comment:
why not add the `_` prefix? I prefer to use `index` over `_index`, `index` is more in line with the naming convention of `cJSON`

Reply:
while compile, it promt the index was a function in string.h
so use _index would be better?

Reply:
what's your compiler? I have never met this before...

Reply:
gcc-4.4.7  centios6.5.
we include the string.h file ,and it has a function namd "index",we could modify the compile option, and ignore it, but  i don't think it is a good idea

Reply:
i use same compile option,such as  Werror, Wall, in my project.

Reply:
that's strange, I don't think a local variable name will conflict with a function name, and I have tested it in my macOS and ubuntu, still can't reproduce your case.
could you show your snippet code?

Reply:
/bin/sh ../../libtool --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../../include     -Werror -g -O2 -Wall -W -Wshadow -std=gnu99 -D_GNU_SOURCE -DCHECK_PRINT -MT cJSON.lo -MD -MP -MF .deps/cJSON.Tpo -c -o cJSON.lo cJSON.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../../include -Werror -g -O2 -Wall -W -Wshadow -std=gnu99 -D_GNU_SOURCE -DCHECK_PRINT -MT cJSON.lo -MD -MP -MF .deps/cJSON.Tpo -c cJSON.c  -fPIC -DPIC -o .libs/cJSON.o
cc1: warnings being treated as errors
cJSON.c: In function ‘get_array_item’:
cJSON.c:1759: error: declaration of ‘index’ shadows a global declaration
/usr/include/string.h:489: error: shadowed declaration is here
cJSON.c: In function ‘cJSON_GetArrayItem’:
cJSON.c:1778: error: declaration of ‘index’ shadows a global declaration
/usr/include/string.h:489: error: shadowed declaration is here

Reply:
we could modify the gcc option to avoid it,but i thought change the variable name would be better.

Reply:
1759 static cJSON* get_array_item(const cJSON *array, size_t index)
   1760 {
   1761     cJSON *current_child = NULL;
   1762 
   1763     if (array == NULL)
   1764     {
   1765         return NULL;
   1766     }
   1767 
   1768     current_child = array->child;
   1769     while ((current_child != NULL) && (index > 0))
   1770     {
   1771         index--;
   1772         current_child = current_child->next;
   1773     }
   1774 
   1775     return current_child;
   1776 }
   1777 
   1778 CJSON_PUBLIC(cJSON *) cJSON_GetArrayItem(const cJSON *array, int index)
   1779 {
   1780     if (index < 0)
   1781     {
   1782         return NULL;
   1783     }
   1784 
   1785     return get_array_item(array, (size_t)index);
   1786 }

Reply:
you could try cmd :
gcc  -I. -Werror -g -O2 -Wall -W -Wshadow -std=gnu99  -c cJSON.c -o cJSON.o

Reply:
the option -Wshadow

Reply:
@xiaomianhehe I run the command you offered, but no warning or error message prompt. I don't refuse to change variable names, but I hope to merge this PR after I figure out the situation.
* First, `string.h` has been introduced into the cJSON.c file, but the warning you mentioned did not appear when compiling cJSON with cmake(cmake has configured the `-Wshadow` option);
* Second, the **local variable** defined in the method will be shadowed only when there is **a global variable** with the same name. Since there is no global variable `index` defined in string.h, so this warning should not occur I think.
so could you please check if it is caused by other reasons?

Reply:
cmakelist file has configure the option "-std=c89" ,but i was using std=gun99
you can try modify the "std=" option in the cmakelist.txt

Reply:
@xiaomianhehe I have modified the `std` to `gnu99`, but still no warning. I tested it in macOS, centOS(7.2, gcc4.8.5) and ubuntu(18.04, gcc 7.4.0).
only when `string.h` has a global varialble named `index`, the prompt will show.  
**Have you changed the file of `/usr/include/string.h`?** make sure there is no redundant global  variable of `index`.
* dup_name.c
```C
#include <stdio.h>
#include <string.h>
#include "lib1.h" // when including lib1.h, it failed.

void func(int index) {
    printf("index = %d\n", index);
}

int main() {
    func(5); // print: index = 5
    char *str = "hello world";
    char *res = index(str, 'o');
    printf("%s\n", res);   // print: o world
    return 0;
}
```
* lib1.h
```C
int index = 0;
```
As far as I know, method parameter of local variable named `index` is very common. such as jsoncpp, you could try to compile it and above example code.

After this issue is resolved, I will merge your PR.

Reply:
You are right. i try gcc-4.4.7(Centos6.5)and gcc-4.8.5(centos7.2), the result was different.
it has warning with gcc-4.4.7.

Reply:
so I think there is no need to change `index`, do you agree?

Reply:
Ok, I revert it.I modify it in my project, because we use old version gcc and centos in our produce env.